src/cmake/Helpers.cmake

```diff
@@ -162,26 +162,29 @@ function(alicevision_add_software software_name)
     list(APPEND SOFTWARE_SOURCE "${CMAKE_CURRENT_BINARY_DIR}/${software_name}_version.rc")
   endif()
 
-  add_executable(${software_name} ${SOFTWARE_SOURCE})
+  add_executable(${software_name}_exe ${SOFTWARE_SOURCE})
```

Comment:
not a big fan of this. I imagine it's because of name conflicts.
I think it makes the cmake code less readable, as one expect that the target name given is used to define the target with that name. And then in the code if u need to do something with the target you need to know that a suffix is added. 

Maybe instead of `masking` we can use `segmentation`, as in this case we are more generating the mask by processing the image rather than performing actions with a mask. That does not solve the name conflict but maybe we give  the library a different, more general, broader meaning (eg imageProcessing on the same line as imgproc of opencv)

Reply:
I don't like the idea of having to play with name variantes, just to avoid stupid cmake conflicts.
I agree that the exe suffix is hidden, but we cannot make the alias explicit as the output will get this name. It looks to me as the least worst solution, but I'm open to other suggestions if you have better idea... other than playing with name variantes which is a pain we have for a too long.

Reply:
* i was not suggesting to use alias or variants, i was just arguing that in cv masking and segmentation are different, and here we are doing segmentation (ie generating the mask) rather than masking (performing operations with existing masks, as eg in feature extraction). Nothing to do with the cmake conflict. That being said, the suggestion of having a lib of broader spectrum like imageProcessing (other algo may coming up in the future) would help to avoid the conflict.

* to at least preserve the encapsulation (ie not see the use of suffix `_exe` outside the call to add_software) we can do this for the targets with optional dependencies:
```cmake
set(<targetname>_LINKS aliceVision_sfmData aliceVision_sfmDataIO etc etc)
if(ALICEVISION_HAVE_CCTAG)
  list(APPEND <targetname>_LINKS CCTag::CCTag)
endif()
alicevision_add_software(<targetname>
      SOURCE main_<targetname>.cpp
      FOLDER ${FOLDER_SOFTWARE_PIPELINE}
      LINKS <targetname>_LINKS ...
```
at least everything stays in `alicevision_add_software` and we have not to go back to the target and use `target_link_libraries`. A bit more verbose but it seems cleaner to me. :-)